src/Api/Controllers/ProviderOrganizationsController.cs

```diff
@@ -58,5 +59,31 @@ public async Task Add(Guid providerId, [FromBody]ProviderOrganizationAddRequestM
 
             await _providerService.AddOrganization(providerId, model.OrganizationId, userId, model.Key);
         }
+
+        [HttpPost("")]
+        [SelfHosted(NotSelfHostedOnly = true)]
+        public async Task<ProviderOrganizationResponseModel> Post(Guid providerId, [FromBody]OrganizationCreateRequestModel model)
+        {
+            var user = await _userService.GetUserByPrincipalAsync(User);
+            if (user == null)
+            {
+                throw new UnauthorizedAccessException();
+            }
+
+            if (!_currentContext.ManageProviderOrganizations(providerId))
+            {
+                throw new NotFoundException();
+            }
+
+            var plan = StaticStore.Plans.FirstOrDefault(plan => plan.Type == model.PlanType);
+            if (!(plan is {LegacyYear: null}))
+            {
+                throw new Exception("Invalid plan selected.");
+            }
```

Comment:
Should this kind of validation live in the service layer?

Reply:
Ideally yes. I followed the same pattern as Organization, but might as well look into moving the logic into the service instead.

Reply:
I found the reason why it's in the controllers. It seems the admin panel can still create legacy plans, moving the logic to the service would prevent this.